### PR TITLE
Novel applications: private order book

### PR DESCRIPTION
**Pull request by phklive**
In this PR I propose to add `clob.md` where we describe a novel application that could be built on top of Miden as a private central limit order book.

### CODE REVIEW

**Comment by phklive** [comment]
For  `clob` based your great work, fixed some formatting / typos mistakes. Made the headings / structure coherent with other `novel_applications` and added headings / sections.

@Dominik1999

File: src/novel_applications/defi/clob.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+1. High TPS - builders ask for 30-40 and some even more
+2. Low finality - sub-second finality would be optimal
+3. Quick and cheap order updates - Order updates should be almost instantaneous and free, especially for Market Makers
+
+Nice to have:
+4. Privacy at the base layer - to not reveal a trading strategy 
+5. Matching engine - FIFO/Pro-Rata should also run on-chain to ensure "best execution"
+6. MEV protection - to ensure "best execution"
+
+(Are there more required features? Do we need to cancel the order quickly?)
```
Comment (Dominik1999) [inline src/novel_applications/defi/clob.md]: I think you can delete this 

(Are there more required features? Do we need to cancel the order quickly?)

File: src/novel_applications/defi/clob.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+Polygon Miden offers the required features (1-6) while simultaneously, computational integrity is secured by Ethereum, and liquidity is provided by Ethereum and the Polygon ecosystem. 
+
+Polygon Miden is designed to have high TPS sustainably. Orders can be updated for free and in no time using updateable transactions. Polygon Miden offers different levels of privacy at the base layer. Matching engines can run off-chain, but there can be a zk-proof of correct execution to be verified by the network. 
+    
+For finality, we need to define what we mean by that. 
+
+## A Powerful Order Book on Miden
+    
+Let's collect ideas for implementing a private on-chain order book exchange on Miden. Given the [Miden Architecture](https://0xpolygonmiden.github.io/miden-base/architecture.html), we can design a simple onchain order book exchange.
```
Comment (Dominik1999) [inline src/novel_applications/defi/clob.md]: This link doesn't work anymore, I think you need to link to the new docs https://docs.polygon.technology/miden/miden-base/architecture/overview/

File: src/novel_applications/defi/clob.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+In our simple example:
+- Users shall be able to place a limit order (buy or sell)
+- Users shall be able to place a market order (buy or sell)
+- Users shall be able to take any order 
+- Users shall be able to cancel any order
+- There shall be some form of best execution
+- The order book shall be visible on any web2 interface like a homepage
+
+In Miden, there are [Accounts](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html) and [Notes](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html), see [here](https://polygon.technology/blog/polygon-miden-transaction-model-2). Notes can be updated for free and in no time; therefore, Notes can represent an order.
```
Comment (Dominik1999) [inline src/novel_applications/defi/clob.md]: I think most of those links don't work anymore

File: src/novel_applications/defi/clob.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+- The order book shall be visible on any web2 interface like a homepage
+
+In Miden, there are [Accounts](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html) and [Notes](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html), see [here](https://polygon.technology/blog/polygon-miden-transaction-model-2). Notes can be updated for free and in no time; therefore, Notes can represent an order. 
+    
+![](https://hackmd.io/_uploads/Skavl8sSn.png)
+    
+All Notes with the same `Tag` can be listed in the same order book. Notes carry the asset to sell in their `Vault` - this cannot be updated, and in this example, it is 1 ETH. In addition, Notes have a single executable `Script`. This script is the root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html). Here is where we can encode our order. For example, `0x1234` can mean:
+
+```
+receive( 1 ETH )  # what is in the vault
```
Comment (Dominik1999) [inline src/novel_applications/defi/clob.md:60]: Here you can link to our simple SWAP-note that you have built.

File: src/novel_applications/defi/clob.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+B = {receive 1 ETH && send 1990 DAI to sender}
+```
+    
+### Market Orders Are Slightly More Complex
+
+A market order is an order to buy or sell an asset immediately. This type of order guarantees that the order will be executed but does not guarantee the execution price. 
+
+A naive implementation would be to create a note having a script that queries the current market price. The script could be like `receive( 1 ETH ), send( current_price, note.sender )`. Whereas `current_price` must come from an oracle but it should be easy to compute given that all notes are public. However, there might be attack vectors for nonliquid markets. 
+
+So now, users can place limit and market orders by creating new notes. Those orders can be updated by simply revealing new parts of the script. Those updates don't need to happen on-chain, but, e.g., in any public forum or directly to the operator, and are therefore for free and can happen in no time. And anybody can take orders by simply consuming the respective notes.
```
Comment (Dominik1999) [inline src/novel_applications/defi/clob.md:98]: Sometimes in this text you use `onchain` and sometimes `on-chain`. Let's use always `on-chain`

**Review by Dominik1999 — APPROVED** [review]
LGTM. I left some comments. The current design by KEOM and Composability Labs is a bit more complex overall, but we don't need to reflect that here.

**Pull request merged**